gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Processes raw variant information using header information.
+
+Note that for creating instances of the data objects in this module, there is a
+factory function create_processed_variant. Other than that function, these
+objects should be used as non-mutable in other scopes, hence all mutating
+functions are "private".
+"""
+
+import logging
+
+from collections import defaultdict
+
+from gcp_variant_transforms.beam_io import vcfio
+
+
+_FIELD_COUNT_ALTERNATE_ALLELE = 'A'
+
+
+class ProcessedVariant(object):
+  """A wrapper around the ``Variant`` class with extra functionality.
+
+  Given header file information, this can parse INFO fields that need to be
+  split and attached to alternates. This is not inherited from
+  :class:``vcfio.Variant`` as an encapsulation layer and to prefer composition.
+  """
+
+  def __init__(self, variant):
+    if not isinstance(variant, vcfio.Variant):
+      raise ValueError('Expected an instance of vcfio.Variant.')
+    self._variant = variant
+    self._non_alt_infos = None
+    self._single_alternate_datas = []
+    for a in variant.alternate_bases:
+      self._single_alternate_datas.append(SingleAlternateData(a))
+
+  def __repr__(self):
+    return self._variant.__repr__()
+
+  def get_reference_name(self):
+    return self._variant.reference_name
+
+  def get_start(self):
+    return self._variant.start
+
+  def get_end(self):
+    return self._variant.end
+
+  def get_reference_bases(self):
+    return self._variant.reference_bases
+
+  def get_names(self):
+    return self._variant.names
+
+  def get_quality(self):
+    return self._variant.quality
+
+  def get_filters(self):
+    return self._variant.filters
+
+  def get_calls(self):
+    return self._variant.calls
+
+  def get_non_alt_infos(self):
```

Comment:
nit: just `info`, it's cleaner :) We can assume that this returns the top-level info and not per-alt or per-call info.